lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -77,8 +90,107 @@
     vpc_id: vpc-123456
     cidr: 10.0.1.16/28
 
+- name: Create subnet with IPv6 block assigned
+  ec2_vpc_subnet:
+    state: present
+    vpc_id: vpc-123456
+    cidr: 10.1.100.0/24
+    ipv6cidr: 2001:db8:0:102::/64
+
+- name: Remove IPv6 block assigned to subnet
+  ec2_vpc_subnet:
+    state: present
+    vpc_id: vpc-123456
+    cidr: 10.1.100.0/24
+    ipv6cidr: ''
+'''
+
+RETURN = '''
+subnet.id:
```

Comment:
This should `subnet` with type `complex` and `contains:` the rest of the information